mlir/include/mlir/Analysis/Presburger/Expr.h

```diff
@@ -0,0 +1,48 @@
+#ifndef MLIR_ANALYSIS_PRESBURGER_EXPR_H
+#define MLIR_ANALYSIS_PRESBURGER_EXPR_H
+
+#include "mlir/Analysis/Presburger/Set.h"
+
+namespace mlir {
+/// This class implements a data structure that represents a piecewise
+/// Presburger expression. Each piece is define by a domain and an expression
+/// that is applied to points lying in this domain. Domains are represented as
+/// parameteric Presburger sets, expressions by normal Presburger expressions.
```

Comment:
Expressions are affine or quasi-affine, but cannot be piecewise as Presburger expressions can?